### PR TITLE
Verify lineNumber

### PR DESCRIPTION
**Pull request by Arcanemagus**
Verify that the lineNumber provided is not past the end of the buffer provided.

### CODE REVIEW

File: lib/helpers.coffee [hunk 1]
```diff
@@ -57,10 +57,12 @@ module.exports = Helpers =
         spawnedProcess.process.stdin.end() # We have to end it or the programs will keep waiting forever
 
   rangeFromLineNumber: (textEditor, lineNumber, colStart) ->
-    throw new Error('Provided text editor is invalid') unless textEditor?.getText?
+    throw new Error('Provided text editor is invalid') unless textEditor?.getBuffer?
     if typeof lineNumber isnt 'number' or lineNumber isnt lineNumber or lineNumber < 0
       return [[0, 0], [0, 1]]
-    lineLength = textEditor.getBuffer().lineLengthForRow(lineNumber)
+    buffer = textEditor.getBuffer()
+    throw new Error('Line number greater than maximum line') if lineNumber > buffer.getLineCount() - 1
```
Comment (steelbrain) [inline lib/helpers.coffee]: It would make it easier to debug if we also logged the given line number and max line number to console in this kinda case

Comment (Arcanemagus) [inline lib/helpers.coffee]: Sure, only didn't because we aren't for column.

Comment (Arcanemagus) [inline lib/helpers.coffee]: Made the colStart one log the values too while I was at it.

**Comment by steelbrain** [comment]
Merge at will

**Pull request merged**